superset/assets/src/visualizations/core/models/ChartMetadata.js

```diff
@@ -2,14 +2,22 @@ export default class ChartMetadata {
   constructor({
     name,
     credits = [],
-    description,
-    thumbnail,
+    description = '',
     show = true,
+    canBeAnnotation = [],
+    supportedAnnotations = [],
+    thumbnail,
   }) {
     this.name = name;
     this.credits = credits;
     this.description = description;
-    this.thumbnail = thumbnail;
     this.show = show;
+    this.canBeAnnotation = canBeAnnotation.reduce((prev, type) => {
+      const lookup = prev;
+      lookup[type] = true;
+      return lookup;
+    }, {});
+    this.supportedAnnotations = supportedAnnotations;
```

Comment:
any reason to not store these with the same data structure? (don't have strong opinion, mostly curious how you were thinking about it)

Reply:
I have same concern about storing as different data structure too. The benefit is it is better for performance. 

When a user select annotation type `TIME_SERIES`, the ui need to loop through all vis to figure out which vis can be used as annotation source for `TIME_SERIES` in addition to `NATIVE` source. An object can provide immediate lookup with better perf than array. Although it is a very small array so doesn't really matter much.